articles/cross_compilation_build_tools_design.md

```diff
@@ -0,0 +1,248 @@
+---
+layout: default
+title: Cross-compiling ROS 2 packages
+abstract:
+  This article is a design proposal for extending the build tools for ROS 2 to
+  simplify cross-compiling ROS 2 packages for platforms currently not supported
+  by ROS 2, or building on a host with different platforms than the target
+  platform. That later use case is particularly useful for speeding up build
+  times by building the code on a powerful development workstation, and then
+  deploying the binaries to a constrained resources hardware that uses the
+  target platform, thus making the development cycle much faster.
+author: >
+  [Thomas Moulard](https://github.com/thomas-moulard),
+  [Juan Rodriguez Hortala](https://github.com/juanrh)
+published: false
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Background
+
+The [cross-compilation tutorial](https://index.ros.org/doc/ros2/Tutorials/Cross-compilation)
+contains instructions for cross-compiling ROS 2 to unsupported platforms, using
+some configuration files for CMake and Docker that is setting maintained in the
+[ros2/cross_compile repository](https://github.com/ros2/cross_compile/).
+However, that tutorial doesn't provide instructions to build other ROS 2
+packages, but just for building a whole ROS 2 distribution. Ideally, we should
```

Comment:
The "Cross-compiling against a pre-built ROS2" section aims to provide an example on how to CC a single ros2 package without building the full distribution.

Reply:
I'll remove that statement, and make it clear that this proposal only contributes build tools to make it easier to use the existing instructions